### ISSUE TITLE
Can this be replaced by webassembly version?

### ISSUE THREAD

**Issue by lygstate** [websockets/bufferutil#148]

**Comment by lpinca** [comment]
Yes, probably, but it must be faster otherwise it does not make sense.

**Comment by lygstate** [comment]
> Yes, probably, but it must be faster otherwise it does not make sense.

It's make sense when I wanna webpack ws

**Comment by lpinca** [comment]
You can webpack native modules.

**Comment by lygstate** [comment]
> You can connect native modules with webpack.

The webpack result are not portable, for example, it's can not running under both windows/linux

**Comment by lpinca** [comment]
> The webpack result are not portable, for example, it's can not running under both windows/linux/osx

I think you can produce different bundles picking the correct pre-compiled binary and use the correct bundle at runtime. I see your point but I'm not sure if it is a common use case.

**Comment by lygstate** [comment]
> > The webpack result are not portable, for example, it's can not running under both windows/linux/osx
> 
> I think you can produce different bundles picking the correct prebuilt binary and use the correct bundle at runtime. I see your point but I'm not sure if it is a common use case.

cross-compiling is a nightmare, for example for windows, we have MSVC,GNU,Electron,
for Linux we have MUSL,GLIBC and different architecture

**Comment by lpinca** [comment]
There are already prebuilt binaries for Windows, macOS, and Linux (glibc) that cover 99% of use cases. Anyway feel free to open a PR.

**Comment by jerch** [comment]
To me this seems to be a perfect wasm-simd candidate.

**Comment by ChALkeR** [comment]
The js version can be made _significantly_ faster, if needed.
This likely doesn't need a webassembly version
